### PR TITLE
node-build compilation fallback

### PR DESCRIPTION
**Pull request by Schniz**
Add the ability to add `--force-arch` to `fnm env` so if a prebuilt version is not found, fnm will use `node-build` to compile the Node.js version locally.

### CODE REVIEW

File: src/compile_node.rs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    },
+}
+
+pub fn compile_node_with_node_build(config: &FnmConfig, version: &Version) -> Result<(), Error> {
+    let path = {
+        let mut path = version.installation_path(config);
+        path.pop();
+        path
+    };
+    let temp_installations_dir = tempfile::tempdir().unwrap();
```
Comment (Schniz) [inline src/compile_node.rs:37]: node-build does not support spaces in directories too. We need to find a solution for that 🥲 
related: 
* https://github.com/Schniz/fnm/issues/622#issuecomment-1029320579

Comment (Schniz) [reply src/compile_node.rs:37]: I mean, the solution is to have a `dirs` module (which we already have) and probably store stuff in `.local/fnm` or something unless XDG_ env vars are provided, but I want to make the migration fast 🥲

**Comment by trm217** [comment]
@Schniz Could you need help working on this? I am currently having a bit of an issue with other team members using nvm and arm64 NodeV14, I would certainly look forward not having to switch back to nvm 😂

**Comment by Schniz** [comment]
I think it’s fairly finished. I’ll take another look and will release it on the next version. The only problem is node-build not supporting spaces in the installation path

**Pull request closed without merging**